src/docs/development/ui/advanced/focus.md

```diff
@@ -0,0 +1,550 @@
+```dart
+class OrderedButtonRow extends StatelessWidget {
+  const OrderedButtonRow({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FocusTraversalGroup(
+      policy: OrderedTraversalPolicy(),
+      child: Row(
+        children: <Widget>[
+          const Spacer(),
+          FocusTraversalOrder(
+            order: NumericFocusOrder(2.0),
+            child: TextButton(child: const Text('ONE'), onPressed: (){}),
+          ),
+          const Spacer(),
+          FocusTraversalOrder(
+            order: NumericFocusOrder(1.0),
+            child: TextButton(child: const Text('TWO'), onPressed: (){}),
+          ),
+          const Spacer(),
+          FocusTraversalOrder(
+            order: NumericFocusOrder(3.0),
+            child: TextButton(child: const Text('THREE'), onPressed: (){}),
+          ),
+          const Spacer(),
+        ],
+      ),
+    );
+  }
+}
+```
+
+### FocusTraversalPolicy
+
+The `FocusTraversalPolicy` is the object that determines which widget is next,
+given a request and the current focus node. The requests (member functions) are
+things like `findFirstFocus`, `findLastFocus`, `next`, `previous`, and
+`inDirection`.
+
+`FocusTraversalPolicy` is the abstract base class for concrete policies, like
+`ReadingOrderTraversalPolicy`,  `OrderedTraversalPolicy` and the
+[`DirectionalFocusTraversalPolicyMixin`][] classes.
+
+In order to use a `FocusTraversalPolicy`, you give one to a
+`FocusTraversalGroup`, which determines the widget subtree in which the policy
+will be effective. The member functions of the class are rarely called directly:
+they are meant to be used by the focus system.
+
+## The focus manager
+
+The [`FocusManager`][] maintains the current primary focus for the system. It
+only has a few pieces of API that are useful to users of the focus system. One
+is the `FocusManager.instance.primaryFocus` property, which contains the
+currently focused focus node and is also accessible from the global
+`primaryFocus` field.
+
+Other useful properties are `FocusManager.instance.highlightMode` and
+`FocusManager.instance.highlightStrategy`. These are used by widgets that need
+to switch between a "touch" mode and a "traditional" (mouse and keyboard) mode
+for their focus highlights. When a user is using touch to navigate, the focus
+highlight is usually hidden, and when they switch to a mouse or keyboard, the
+focus highlight needs to be shown again so they know what is focused. The
+`hightlightStrategy` tells the focus manager how to interpret changes in the
+usage mode of the device: it can either automatically switch between the two
+based on the most recent input events, or it can be locked in touch or
+traditional modes. The provided widgets in Flutter already know how to use this
+information, so you only need it if you're writing your own controls from
+scratch. You can use `addHighlightModeListener` callback to listen for changes
+in the highlight mode.
```

Comment:
```suggestion
`FocusManager.instance.highlightMode` and `FocusManager.instance.highlightStrategy`
are also useful. Widgets use these properties
to switch between "touch" and "traditional" (mouse and keyboard) modes
for focus highlights. When a user navigates by touch, the focus
highlight is usually hidden, and when switching to a mouse or keyboard,
the focus highlight indicates what has focus.
The `highlightStrategy` tells the focus manager how to interpret changes in the
usage mode for the device: it can either automatically switch between the two
based on the most recent input events, or it can be locked in touch or
traditional modes. The provided widgets in Flutter already know how to use this
information, so you only need it when writing your own controls from scratch.
Use the `addHighlightModeListener` callback to listen for changes to the highlight mode.
```
Why would you ever lock focus to keyboard or touch? It seems very unfriendly to those
with accessibility issues.